apps/src/music/views/AdvancedControls.tsx

```diff
@@ -0,0 +1,146 @@
+import classNames from 'classnames';
+import Checkbox from '@cdo/apps/componentLibrary/checkbox/Checkbox';
+import {MAX_BPM, MIN_BPM} from '../constants';
+
+/**
+ * Displays advanced controls used only by the ToneJS player, including
+ * custom BPM, custom key, and loop settings.
+ *
+ * Currently this is internal-only.
+ */
+const AdvancedControls: React.FunctionComponent = () => {
+  const dispatch = useAppDispatch();
+  const bpm = useAppSelector(state => state.music.bpm);
+  const key = useAppSelector(state => state.music.key);
+  const loopEnabled = useAppSelector(state => state.music.loopEnabled);
+  const loopStart = useAppSelector(state => state.music.loopStart);
+  const loopEnd = useAppSelector(state => state.music.loopEnd);
+  const isPlaying = useAppSelector(state => state.music.isPlaying);
+
+  const onBpmChange = useCallback(
+    (bpm: number) => {
+      dispatch(setBpm(bpm));
+    },
+    [dispatch]
+  );
+
+  const onKeyChange = useCallback(
+    (key: string) => {
+      dispatch(setKey(Key[key as keyof typeof Key]));
+    },
+    [dispatch]
+  );
+
+  const onLoopChange = useCallback(
+    (checked: boolean) => {
+      dispatch(setLoopEnabled(checked));
+    },
+    [dispatch]
+  );
+
+  const onLoopStartChange = useCallback(
+    (loopStart: number) => {
+      dispatch(setLoopStart(loopStart));
+    },
+    [dispatch]
+  );
+
+  const onLoopEndChange = useCallback(
+    (loopEnd: number) => {
+      dispatch(setLoopEnd(loopEnd));
+    },
+    [dispatch]
+  );
+
+  return (
+    <div className={moduleStyles.container}>
+      <Heading5 className={moduleStyles.title}>Advanced Controls</Heading5>
+      <div className={moduleStyles.section}>
+        <div className={moduleStyles.row}>
+          <div>BPM: {bpm}</div>
+        </div>
+        <input
+          type="range"
+          min={MIN_BPM}
+          max={MAX_BPM}
+          step="5"
+          value={bpm}
+          onChange={event => {
+            onBpmChange(Number(event.target.value));
+          }}
+          disabled={isPlaying}
+        />
+      </div>
+      <div className={classNames(moduleStyles.row, moduleStyles.section)}>
+        <div className={moduleStyles.rowLabel}>Key</div>
+        <select
+          className={moduleStyles.input}
+          name="key"
+          value={Key[key]}
+          onChange={event => {
+            onKeyChange(event.target.value);
+          }}
+          disabled={isPlaying}
+        >
+          {Object.keys(Key)
+            .filter(key => isNaN(Number(key)))
+            .map((key, index) => {
+              return <option key={index}>{key}</option>;
+            })}
+        </select>
+      </div>
+      <div className={classNames(moduleStyles.section)}>
+        <div className={moduleStyles.row}>
+          <div className={moduleStyles.rowLabel}>Loop</div>
+          <Checkbox
+            name="loop"
+            checked={loopEnabled}
+            onChange={event => onLoopChange(event.target.checked)}
+          />
```

Comment:
It would be nice if this component supported a white border, but the current dark border is alright for internal use for now.